File: packages/@cdktf/cli-core/templates/go/main_test.go
package main

import (
	"github.com/aws/jsii-runtime-go"
	"github.com/hashicorp/terraform-cdk-go/cdktf"
	"testing"
)

// The tests below are example tests, you can find more information at
// https://cdk.tf/testing

/*
var stack = NewMyApplicationsAbstraction(cdktf.Testing_App(nil), "stack")
var synth = cdktf.Testing_Synth(stack)

func TestShouldContainContainer(t *testing.T){
	assertion := cdktf.Testing_ToHaveResource(synth, docker.Container_TfResourceType())

	if !*assertion  {
		t.Error("Assertion Failed")
	}
}

func TestShouldUseUbuntuImage(t *testing.T){
	properties := map[string]interface{}{
		"name": "ubuntu:latest",
	}
	assertion := cdktf.Testing_ToHaveResourceWithProperties(synth, docker.Image_TfResourceType(), &properties)

	if !*assertion  {
		t.Error("Assertion Failed")
	}
}

func TestCheckValidity(t *testing.T){
	assertion := cdktf.Testing_ToBeValidTerraform(cdktf.Testing_FullSynth(stack))

	if !*assertion  {
		t.Error("Assertion Failed")
	}
}
*/

File: packages/@cdktf/cli-core/templates/python/main-test.py
import pytest
from cdktf import Testing

# The tests below are example tests, you can find more information at
# https://cdk.tf/testing


class TestMain:

    def test_my_app(self):
        assert True

    #stack = TerraformStack(Testing.app(), "stack")
    #app_abstraction = MyApplicationsAbstraction(stack, "app-abstraction")
    #synthesized = Testing.synth(stack)

    # def test_should_contain_container(self):
    #    assert Testing.to_have_resource(self.synthesized, Container.TF_RESOURCE_TYPE)

    # def test_should_use_an_ubuntu_image(self):
    #    assert Testing.to_have_resource_with_properties(self.synthesized, Image.TF_RESOURCE_TYPE, {
    #        "name": "ubuntu:latest",
    #    })

    # def test_check_validity(self):
    #    assert Testing.to_be_valid_terraform(Testing.full_synth(stack))
